meson.build

```diff
@@ -0,0 +1,184 @@
+if flatpak_intf_dir == ''
+    flatpak_dep = dependency('flatpak', version: '>= 1.5.0')
+    flatpak_intf_dir = flatpak_dep.get_variable(pkgconfig: 'interfaces_dir')
+endif
+
+systemd_userunit_dir = get_option('systemd-user-unit-dir')
+if systemd_userunit_dir == ''
+    # This is deliberately not ${libdir}: systemd units always go in
+    # .../lib, never .../lib64 or .../lib/x86_64-linux-gnu
+    systemd_userunit_dir = prefix / 'lib' / 'systemd' / 'user'
+endif
+
+dataroot_dir = get_option('datarootdir')
+if dataroot_dir == ''
+    dataroot_dir = datadir
+endif
+
+installed_tests_dir = prefix / libexecdir / 'installed-tests' / meson.project_name()
+installed_tests_data_dir = prefix / datadir / 'installed-tests' / meson.project_name()
+docs_dir = datadir / 'doc' / meson.project_name()
+
+summary({
+	'DBus service dir': dbus_service_dir,
+	'Flatpak interfaces dir': flatpak_intf_dir,
+	'systemd user unit dir': systemd_userunit_dir,
+	'Installed tests dir': installed_tests_dir,
+	},
+	section: 'Directories',
+)
+
+###### various include directories we'll use later
+# These are set here so meson handles the relative paths correctly,
+# makes life easier for us
+
+common_includes = include_directories('.')  # config.h
+src_includes = include_directories('src')
+
+###### plugins, dependencies, compiler setup
+
+i18n = import('i18n')
+gnome = import('gnome')
+pkgconfig = import('pkgconfig')
+
+cc = meson.get_compiler('c')
+cflags = [
+    '-Wno-unused-parameter',
+    '-Wno-sign-compare',
+    '-Wno-missing-field-initializers',
+]
+add_project_arguments(cc.get_supported_arguments(cflags), language : 'c')
+
+config_h = configuration_data()
+config_h.set('_GNU_SOURCE', 1)
+config_h.set_quoted('G_LOG_DOMAIN', 'xdg-desktop-portal')
+config_h.set_quoted('DATADIR', datadir)
+config_h.set_quoted('LIBEXECDIR', libexecdir)
+config_h.set_quoted('LOCALEDIR', localedir)
+config_h.set_quoted('GETTEXT_PACKAGE', 'xdg-desktop-portal')
+config_h.set_quoted('PACKAGE_STRING', 'xdg-desktop-portal @0@'.format(meson.project_version()))
+if cc.has_function('renameat2')
+  config_h.set('HAVE_RENAMEAT2', 1)
+endif
+
+glib_dep = dependency('glib-2.0')
+gio_dep = dependency('gio-2.0')
+gio_unix_dep = dependency('gio-unix-2.0')
+json_glib_dep = dependency('json-glib-1.0')
+fuse3_dep = dependency('fuse3', version: '>= 3.10.0')
+gdk_pixbuf_dep = dependency('gdk-pixbuf-2.0')
+geoclue_dep = dependency('libgeoclue-2.0',
+                         version: '>= 2.5.2',
+                         required: get_option('geoclue'))
+libportal_dep = dependency('libportal',
+                           required: get_option('libportal'))
+pipewire_dep = dependency('libpipewire-0.3',
+                          version: '>= 0.2.90',
+                          required: get_option('pipewire'))
+libsystemd_dep = dependency('libsystemd', required: get_option('systemd'))
+
+bwrap = find_program('bwrap')
+config_h.set_quoted('BWRAP', bwrap.full_path())
```

Comment:
Is this used anywhere? I do not see it in the code.

Reply:
no, apparently not. It's only used as compiler flag to `xdg-desktop-portal-validate-icon` but the `config.h` entry is not necessary. Please file a PR to remove it, thanks.

Reply:
Done in https://github.com/flatpak/xdg-desktop-portal/pull/860